quinn-proto/src/connection/datagrams.rs

```diff
@@ -147,6 +153,7 @@ impl DatagramState {
         }
 
         self.outgoing_total -= datagram.data.len();
+        self.send_blocked = false;
```

Comment:
We should reset this after emitting an event, rather than only on the next write call. That way we guarantee that redundant events won't be emitted.